mlos_bench/mlos_bench/launcher.py

```diff
@@ -227,6 +230,20 @@ def _load_config(self,
             global_config["config_path"] = config_path
         return global_config
 
+    def _expand_vars(self, value: Any) -> Any:
+        """
+        Expand dollar variables in the globals.
```

Comment:
First, thanks for this!
I had a thought about this during some local hacking I was doing along the same lines.
What do you think about doing this recursively until all `$variables` are expanded or can't be (in which case, throw an error)?
The idea is basically, to support things like this, where one value might also include the results of a template from another one:

- template vmServerName with `"vmServerName": "vmServer-$experiment-$location"`
- template vnetName with `"vnetName": "vnet-$vmServerName"`

Reply:
yes, we process the parameters in the order they appear in config. E.g., in the following fragment
```python
{
    "experiment_id": "RedisBench",
    "location": "westus2",
    "vmServerName": "vmServer-$experiment_id-$location",
    "vnetName": "vnet-$vmServerName",
}
```
`vnetName` would expand to `vnet-vmServer-RedisBench-westus2`

Reply:
Ah, if the order manages it, then it should be ok I think.  Thanks!  That's how arm templates work too I believe.

Reply:
Fixed in #493